crates/rpc/src/traits.rs

```diff
@@ -0,0 +1,65 @@
+use crate::api::SupervisorApiClient;
+use alloc::{boxed::Box, vec::Vec};
+use alloy_primitives::Log;
+use alloy_sol_types::SolEvent;
+use async_trait::async_trait;
+use core::time::Duration;
+use maili_interop::{ExecutingMessage, SafetyLevel, CROSS_L2_INBOX_ADDRESS};
+use tokio::time::error::Elapsed;
+
+/// Failures occurring during validation of [ExecutingMessage]s.
+#[derive(thiserror::Error, Debug)]
+pub enum ExecutingMessageValidatorError {
+    /// Failure during Supervisor's validation of [ExecutingMessage]s.
+    #[error("Supervisor determined messages are invalid: {0}")]
+    SupervisorValidationError(#[from] jsonrpsee_core::ClientError),
```

Comment:
as a separate issue, we may want to create an enum to parse returned rpc errors from the supervisor into, but `jsonrpsee_core::ClientError` is good for now

Reply:
Feel free to @ me in any related / followup issues 👍